### PR TITLE
(maint) update README with learnings about composite project behavior

### PR DESCRIPTION
**Pull request by jonathannewman**
This updates the README with some information about behvior that wasn't necessarily obvious, and helpful for composite projects.

Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,9 +134,24 @@ Note that the symble `:ezbake` is not strictly necessary here.
...
+above.
+
+If the composite project needs to define build time templates, such as `resources/ext/cli_defaults/cli-defaults.sh.erb`
+EZbake will only include items that are in jars on the class path at the time
+the project is staged. As a result, the project must include a self-referential dependency
+in the `ezbake` profile. This will allow ezbake to discover these files correctly.
+```clojure
+(def my-project-version "0.0.1-SNAPSHOT")
+
+(defproject somenamespace/myproject my-project-version
```
Comment (justinstoller) [inline README.md:146]:
```suggestion
(defproject somenamespace/myproject ~my-project-version
```

**Review by justinstoller — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jonathannewman) [reply README.md:146]: I looked at our other uses of composite projects, and none of them use the tilde in the version declaration section of `defproject` Other uses of the version within the map in `defproject` are tilde expanded.

**Comment by steveax** [comment]
Actually, looks like the `~` should only be present when used in deps?

**Comment by justinstoller** [comment]
bizarre

**Pull request merged**